### PR TITLE
scss, pre-compile jsx, cleaner watch script, working test coverage, precommit hooks

### PR DESCRIPTION
**Pull request by will-weiss**
A number of quality of life improvements, the output of the build script is far more readable and restarts much more quickly.

Also adds ahead of time compilation of scss and the jsx templates.

Also adds code coverage

```
$ npm run dev

> @ dev /Users/willweiss/dev/morehumaninternet/roar-server
> NODE_ENV=dev npm run build -- --watch


> @ build /Users/willweiss/dev/morehumaninternet/roar-server
> npm run clear && node build.js "--watch"


> @ clear /Users/willweiss/dev/morehumaninternet/roar-server
> concurrently 'rimraf bundled' 'rimraf css' 'rimraf html'

[0] rimraf bundled exited with code 0
[1] rimraf css exited with code 0
[2] rimraf html exited with code 0
🛠  Building the Roar! server in watch mode 

typescript 2021-01-05 21:01:39.557 
typescript 2021-01-05 21:01:39.557 9:01:39 PM - Starting compilation in watch mode...
typescript 2021-01-05 21:01:39.557 
views 2021-01-05 21:01:39.641 
views 2021-01-05 21:01:39.641 > @ views /Users/willweiss/dev/morehumaninternet/roar-server
views 2021-01-05 21:01:39.641 > node views.js "--watch"
views 2021-01-05 21:01:39.641 
scss:index 2021-01-05 21:01:39.743 Compiled scss/index.scss to css/index.css.
scss:index 2021-01-05 21:01:39.743 Sass is watching for changes. Press Ctrl-C to stop.
scss:index 2021-01-05 21:01:39.743 
scss:common 2021-01-05 21:01:39.785 Compiled scss/common.scss to css/common.css.
scss:common 2021-01-05 21:01:39.786 Sass is watching for changes. Press Ctrl-C to stop.
scss:common 2021-01-05 21:01:39.786 
scss:welcome 2021-01-05 21:01:39.838 Compiled scss/welcome.scss to css/welcome.css.
scss:welcome 2021-01-05 21:01:39.839 Sass is watching for changes. Press Ctrl-C to stop.
scss:welcome 2021-01-05 21:01:39.839 
views 2021-01-05 21:01:40.076 Compiling ./views/index.jsx
views 2021-01-05 21:01:40.388 Compiling ./views/welcome.jsx
views 2021-01-05 21:01:40.393 Compiled .jsx files in /views to /html
typescript 2021-01-05 21:01:42.708 
typescript 2021-01-05 21:01:42.708 9:01:42 PM - Found 0 errors. Watching for file changes.

🎉 First build successful! Continuing to watch files...


> @ start /Users/willweiss/dev/morehumaninternet/roar-server
> node -r source-map-support/register compiled/entry.js

roar-server listening on 5004 Tue Jan 05 2021 21:01:43 GMT-0500 (Eastern Standard Time)
Established connection to redis
typescript 2021-01-05 21:02:39.217 
typescript 2021-01-05 21:02:39.217 9:02:39 PM - File change detected. Starting incremental compilation...
typescript 2021-01-05 21:02:39.217 
typescript 2021-01-05 21:02:39.298 
typescript 2021-01-05 21:02:39.298 9:02:39 PM - Found 0 errors. Watching for file changes.
restarting server...

> @ start /Users/willweiss/dev/morehumaninternet/roar-server
> node -r source-map-support/register compiled/entry.js

roar-server listening on 5004 Tue Jan 05 2021 21:02:40 GMT-0500 (Eastern Standard Time)
Established connection to redis
typescript 2021-01-05 21:07:17.277 
typescript 2021-01-05 21:07:17.277 9:07:17 PM - File change detected. Starting incremental compilation...
typescript 2021-01-05 21:07:17.277 
typescript 2021-01-05 21:07:17.343 
typescript 2021-01-05 21:07:17.343 9:07:17 PM - Found 0 errors. Watching for file changes.
restarting server...

> @ start /Users/willweiss/dev/morehumaninternet/roar-server
> node -r source-map-support/register compiled/entry.js

roar-server listening on 5004 Tue Jan 05 2021 21:07:18 GMT-0500 (Eastern Standard Time)
Established connection to redis
```

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "arrowParens": "avoid",
+  "singleQuote": true,
+  "semi": false,
+  "printWidth": 160
```
Comment (shachar-langer) [inline .prettierrc:5]: 160 is much higher than the recommended 80. I checked the I can fit 120 characters on my 13-inch screen. What do you think about setting it to 120?

File: package.json [hunk 2]
```diff
@@ -1,16 +1,16 @@
 {
   "scripts": {
-    "clean": "rm -rf compiled",
-    "compile": "tsc",
-    "compile:watch": "tsc -w",
-    "dev": "npm run clean && npm run compile && { npm run compile:watch &\n NODE_ENV=dev nodemon &\n wait\n }",
+    "build": "npm run clear && node build.js",
+    "build:ensure": "{ [ -d compiled ] && [ -d css ] && [ -d html ] ; } || npm run build",
```
Comment (shachar-langer) [inline package.json:4]: Why are you checking if these folders exist before building?

File: scss/index.scss [hunk 3]
```diff
@@ -114,34 +118,53 @@
...
 }
 
-.acc-item.active .panel {
-  display: block;
+.accordion_item:not(.active) .accordion__text {
+  display: none;
+}
+
+.arrow {
+  transition: all 0.2s;
```
Comment (shachar-langer) [inline scss/index.scss]: Specifying the property we are transitioning improves performance. Please replace `all` with `transform`

File: views/shared/layout.jsx [hunk 4]
```diff
@@ -11,7 +11,7 @@ module.exports = function Layout(props) {
   return (
     <html lang="en">
       <head>
-        <meta charset="UTF-8" />
+        <meta charSet="UTF-8" />
```
Comment (shachar-langer) [inline views/shared/layout.jsx:14]: What is the difference between `charset` and `charSet`? I think HTML is case-insensitive

File: views/shared/layout.jsx [hunk 5]
```diff
@@ -11,7 +11,7 @@ module.exports = function Layout(props) {
   return (
```
Comment (shachar-langer) [inline views/shared/layout.jsx:11]: In line 28 of this file, we add a link of `buttons.css` but this file doesn't exist anymore and it causes an error in the console.

File: views/index.jsx [hunk 6]
```diff
@@ -25,7 +25,7 @@ function Modal() {
     <section className="modal modal--hidden">
       <div className="modal__content">
         <div className="modal__video">
-          <iframe src="https://www.youtube.com/embed/QH2-TGUlwu4" width="560" height="315" frameborder="0"></iframe>
+          <iframe src="https://www.youtube.com/embed/QH2-TGUlwu4" width="560" height="315" frameBorder="0"></iframe>
```
Comment (shachar-langer) [inline views/index.jsx:28]: What is the difference between `frameborder` and `frameBorder`? I think HTML is case-insensitive

File: .prettierrc (hunk 1 above)
Comment (will-weiss) [reply .prettierrc:5]: We use 160 on the other project, I think it's fine.


https://www.theregister.com/2020/06/01/linux_5_7/

File: package.json (hunk 2 above)
Comment (will-weiss) [reply package.json:4]: A common workflow is to have the dev server running, but then to also want to run the tests.

File: views/shared/layout.jsx (hunk 4 above)
Comment (will-weiss) [reply views/shared/layout.jsx:14]: React warnings told me to

File: views/shared/layout.jsx (hunk 5 above)
Comment (will-weiss) [reply views/shared/layout.jsx:11]: I'll fix that up.

File: env.js [hunk 7]
```diff
@@ -1,5 +1,11 @@
+const dotenv = require('dotenv')
+
 if (!process.env.NODE_ENV) {
   throw new Error('Must set NODE_ENV')
 }
 
-require('dotenv').config({ path: `.env.${process.env.NODE_ENV}` })
+if (process.env.NODE_ENV !== 'prod') {
+  dotenv.config({ path: `.default.env` })
```
Comment (shachar-langer) [inline env.js]: I don't think we should use `.default.env` in our code. People will change this file and commit real credentials. Once the credentials are on GitHub it's really hard to erase them. It should be an example file. Maybe rename it to `.env.example`.

**Review by shachar-langer — APPROVED** [review]

**Pull request merged**